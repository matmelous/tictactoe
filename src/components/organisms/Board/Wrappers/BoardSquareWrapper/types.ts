import { UpdateGameType } from "../../../../../state/game/types";
import { MoveType, SquaresType } from "../../../../../types/types";

export type PropsType = {
  id: number;
};

export type OnClickType = {
  id: number;
  squares: SquaresType;
  currentMove: MoveType;
  stateDispatcherFunction: (payload: UpdateGameType) => void;
  playerOne: string;
  playerTwo: string;
};
