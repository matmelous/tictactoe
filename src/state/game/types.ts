import { playerType, squaresType } from "../../types/types";

export type gameSlice = {
  player: playerType;
  player1Name: string;
  player2Name: string;
  winner: null | playerType | boolean;
  squares: squaresType;
  history: {
    player1: number;
    player2: number;
  };
};
