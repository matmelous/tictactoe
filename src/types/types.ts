import { player } from "./enums";

export type playerType = player.cross | player.circle;

export type squaresType = {
  [key: number]: null | playerType;
};
