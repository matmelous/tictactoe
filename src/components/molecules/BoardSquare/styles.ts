import styled from "styled-components";
import Button from "@mui/material/Button";

export const BoardSquareStyled = styled(Button)`
  width: 100px;
  height: 100px;
`;
