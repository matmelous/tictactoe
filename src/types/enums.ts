export enum MoveEnum {
  cross = "x",
  circle = "o",
}
