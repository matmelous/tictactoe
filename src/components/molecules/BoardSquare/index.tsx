import React, { useEffect, useState, useCallback } from 'react';
import { player } from '../../../types/enums';
import Circle from '../../atoms/Circle';
import Cross from '../../atoms/Cross';
import { BoardSquareStyled } from './styles';

type propsType = {
  value: string | null,
  onClick?: () => void
}

const BoardSquare = ({ value, onClick }: propsType) => {
  const [children, setChildren] = useState<null | React.ReactNode>(null);
  const checkValue = useCallback((value: string | null) => {
    switch (value) {
      case null:
        return null;
      case player.cross:
        return <Cross />
      case player.circle:
        return <Circle />
      default:
        return null;
    }
  }, []);

  useEffect(() => {
    setChildren(checkValue(value));
  }, [value, checkValue]);

  return (
    <BoardSquareStyled onClick={onClick}>
      {children}
    </BoardSquareStyled>
  );
}

export default BoardSquare;
