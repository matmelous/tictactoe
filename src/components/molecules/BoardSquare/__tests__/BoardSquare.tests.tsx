import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import BoardSquare from "../index";
import { player } from "../../../../types/enums";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<BoardSquare value={null} disabled={false} />);
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByRole('button')).not.toBeDisabled();
});

it("renders with circle disabled", () => {
  render(<BoardSquare value={player.circle} disabled={true} />);
  expect(screen.getByRole('button')).toBeDisabled();
  expect(screen.getByRole('circle')).toBeInTheDocument();
});

it("renders with cross disabled", () => {
  render(<BoardSquare value={player.cross} disabled={true} />);
  expect(screen.getByRole('button')).toBeDisabled();
  expect(screen.getByRole('cross')).toBeInTheDocument();
});
