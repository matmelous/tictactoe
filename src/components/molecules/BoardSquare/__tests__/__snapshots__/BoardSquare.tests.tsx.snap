// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardSquare renders 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root sc-eCImPb fdDrdY css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="board-square"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
